security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sds
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	authapi "github.com/envoyproxy/go-control-plane/envoy/api/v2/auth"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	sds "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
+	"github.com/gogo/protobuf/types"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+)
+
+// SecretType is used for secret discovery service to construct response.
+const SecretType = "type.googleapis.com/envoy.api.v2.Secret"
+
+// TODO(quanlin): secret struct and secretStore interface are placeholders that used for initial check in.
+// will move them to separated file or use existing structure.
```

Comment:
To be more concrete I think you can use `KeyCertBundle`

Reply:
thanks, that's what I was thinking but not exactly sure for now(we could tweak this part later), added in the comments.